lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,482 @@
+        #        These are the three VII thresholds in kg/m2.
+        self.ice_thresholds = (0.5, 1.0, 2.0)
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=self.radius,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    @staticmethod
+    def _update_metadata(cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+            1. Rename to
+                "probability_of_lightning"
+            2. Remove "threshold" coord
+                (or causes iris.exceptions.CoordinateNotFoundError)
+            3. Discard all cell_methods
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("probability_of_lightning")
+        new_cube.remove_coord('threshold')
+        new_cube.cell_methods = None
+        return new_cube
+
+    def _modify_first_guess(self, cube, first_guess_lightning_cube,
+                            lightning_rate_cube, prob_precip_cube,
+                            prob_vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            first_guess_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube (closest slice in
+                time is used with a maximum time mismatch of 2 hours).
+                This is included to allow this cube to come from a different
+                modelling system, such as the UM.
+
+            lightning_rate_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            prob_precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
```

Comment:
Just to make you aware sphinx doesn't put a newline between these lines when you look at the docs, so it looks like you are missing full stops in your docstings. It's happening a lot in the docstring for this function.